cmd/metrics-v2.go

```diff
@@ -0,0 +1,1189 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/minio/minio/cmd/logger"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// MetricNamespace is top level grouping of metrics to create the metric name.
+type MetricNamespace string
+
+// MetricSubsystem is the sub grouping for metrics within a namespace.
+type MetricSubsystem string
+
+const (
+	bucketMetricNamespace    MetricNamespace = "minio_bucket"
+	clusterMetricNamespace   MetricNamespace = "minio_cluster"
+	healMetricNamespace      MetricNamespace = "minio_heal"
+	interNodeMetricNamespace MetricNamespace = "minio_inter_node"
+	nodeMetricNamespace      MetricNamespace = "minio_node"
+	minIOMetricNamespace     MetricNamespace = "minio"
+	s3MetricNamespace        MetricNamespace = "minio_s3"
+)
+
+const (
+	cacheSubsystem          MetricSubsystem = "cache"
+	capacityRawSubsystem    MetricSubsystem = "capacity_raw"
+	capacityUsableSubsystem MetricSubsystem = "capacity_usable"
+	diskSubsystem           MetricSubsystem = "disk"
+	nodesSubsystem          MetricSubsystem = "nodes"
+	objectsSubsystem        MetricSubsystem = "objects"
+	replicationSubsystem    MetricSubsystem = "replication"
+	requestsSubsystem       MetricSubsystem = "requests"
+	timeSubsystem           MetricSubsystem = "time"
+	trafficSubsystem        MetricSubsystem = "traffic"
+	usageSubsystem          MetricSubsystem = "usage"
+	softwareSubsystem       MetricSubsystem = "software"
+)
+
+// MetricNames are the individual names for the metric.
+type MetricNames string
+
+const (
+	errorsTotal   MetricNames = "error_total"
+	healTotal     MetricNames = "heal_total"
+	hitsTotal     MetricNames = "hits_total"
+	inflightTotal MetricNames = "inflight_total"
+	missedTotal   MetricNames = "missed_total"
+	objectTotal   MetricNames = "object_total"
+	offlineTotal  MetricNames = "offline_total"
+	onlineTotal   MetricNames = "online_total"
+	total         MetricNames = "total"
+
+	failedBytes   MetricNames = "failed_bytes"
+	freeBytes     MetricNames = "free_bytes"
+	pendingBytes  MetricNames = "pending_bytes"
+	receivedBytes MetricNames = "received_bytes"
+	sentBytes     MetricNames = "sent_bytes"
+	totalBytes    MetricNames = "total_bytes"
+	usableBytes   MetricNames = "usable_total_bytes"
```

Comment:
This creates name with repeated `usable` i.e. `minio_cluster_capacity_usable_usable_total_bytes`

Reply:
Fixed